### ISSUE TITLE
Question about subclass claim in the readme

### ISSUE THREAD

**Issue by ljharb** [tabatkins/proposal-item-method#1]
> However, that would mean the values aren't of type Array; various type-checking methods in the community looking for an Array would fail.

Which methods would fail? `instanceof Array` would work, `Array.isArray()` would work; I'm not aware of any idiom or convention for using `x.constructor === Array`, which wouldn't work.

**Comment by tabatkins** [comment]
I don't actually know! That was given to me as a downside by @domenic.

**Comment by domenic** [comment]
The section is generally a bit confused. You can't subclass ObservableArray since it's not a class. It's just a proxy around an array. So the statement "We _could_ address this by subclassing ObservableArray" isn't accurate; the point of the ObservableArray Web IDL construct is to just _be_ an Array, or at least a proxy around one.

**Comment by tabatkins** [comment]
Ah, that makes sense.

In that case, tho, then it should be just as possible to wrap that Proxy around anything that extends Array, yeah?  Maybe we do this automatically, maybe we have an explicit opt-in, but either way, does that work?

**Comment by ljharb** [comment]
A Proxy around an Array could certainly provide this proposal's method for a Get on "item", regardless of what this proposal was named.

**Comment by syg** [comment]
> A Proxy around an Array could certainly provide this proposal's method for a Get on "item", regardless of what this proposal was named.

Yes, the expressivity is certainly allowed. But it is pretty confusing and poor DX since it'd like there's a method being materialized out of thin air, or it'd look like a per-instance own method.

The desire here is so that for the web dev, this new ObservableArray WebIDL type has the same API surface as Arrays. In terms of JS MOP operations, the hope would be that the web dev doesn't need to think about a different Array-like class or subclass. The difference between ObservableArray and Array would be only limited to e.g. stricted types of allowed elements.
